nautobot_device_lifecycle_mgmt/models.py

```diff
@@ -271,6 +407,9 @@ class DeviceSoftwareValidationResult(PrimaryModel):
         blank=True,
         related_name="+",
     )
+    old_software = models.ForeignKey(
+        to="SoftwareLCM", on_delete=models.CASCADE, help_text="Device software", null=True, blank=True, related_name="+"
```

Comment:
Same as above.

```suggestion
        to="SoftwareLCM", on_delete=models.SET_NULL, help_text="Device software", null=True, blank=True, related_name="+"
```

Reply:
Agreed. Let's set it to `models.SET_NULL`